source/common/stats/fake_symbol_table_impl.h

```diff
@@ -142,7 +142,9 @@ class FakeSymbolTableImpl : public SymbolTable {
   }
 
   StoragePtr encodeHelper(absl::string_view name) const {
-    ASSERT(!absl::EndsWith(name, "."));
+    while (absl::EndsWith(name, ".")) {
+      name.remove_suffix(1);
```

Comment:
Instead of this while loop (and elsewhere), what about using `name.find_last_not_of('.')` and removing the appropriate number of suffix characters based on the result?